### PR TITLE
Flexible packet initial radius

### PR DESCRIPTION
**Pull request by andrewfullard**
**Description**

The PacketCollection now includes the initial radius of packets. For the BlackBodySource this is just the inner radius of the ejecta.

**Motivation and context**

Preparation for creating packets without the inner radius set explicitly to the model inner radius e.g. for nebular phase

**How has this been tested?**
- [x] Testing pipeline.
- [ ] Other. 

**Examples**

**Type of change**

- [ ] Bug fix. 
- [x] New feature. 
- [ ] Breaking change. 
- [ ] None of the above. 

**Checklist**

- [x] My change requires a change to the documentation.
    - [ ] I have updated the documentation accordingly.
    - [ ] (optional) I have built the documentation on my fork following [the instructions](https://tardis-sn.github.io/tardis/development/documentation_preview.html).
- [x] I have assigned and requested two reviewers for this pull request.

### CODE REVIEW

**Review by chvogl — APPROVED** [review]
This looks good and sets the groundwork for treating different energy input modes in TARDIS.

File: tardis/montecarlo/base.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ def _initialize_geometry_arrays(self, model):
         self.r_outer_cgs = model.r_outer.to("cm").value
         self.v_inner_cgs = model.v_inner.to("cm/s").value
 
-    def _initialize_packets(self, T, no_of_packets, iteration):
+    def _initialize_packets(self, T, no_of_packets, iteration, radius):
```
[the remark this answers is not in the document]
Comment (andrewfullard) [reply tardis/montecarlo/base.py:169]: Agreed. For the restructuring to come!

File: tardis/montecarlo/base.py [hunk 2]
```diff
@@ -175,10 +175,11 @@ def _initialize_packets(self, T, no_of_packets, iteration):
         seed = self.seed + iteration
         rng = np.random.default_rng(seed=seed)
         seeds = rng.choice(MAX_SEED_VAL, no_of_packets, replace=True)
-        nus, mus, energies = self.packet_source.create_packets(
-            T, no_of_packets, rng
+        radii, nus, mus, energies = self.packet_source.create_packets(
```
[the remark this answers is not in the document]
Comment (andrewfullard) [reply tardis/montecarlo/base.py:178]: Possibly, but we also expect the radius to be different for different packets in the nebular phase. So it needs to be an array of radii corresponding to the energy deposition radius.

File: tardis/montecarlo/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (wkerzendorf) [reply tardis/montecarlo/base.py:169]: can we make this an issues

File: tardis/montecarlo/base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (wkerzendorf) [reply tardis/montecarlo/base.py:178]: I think there are all sorts of potential ways to configure this - but for now I think it's fine how it is.

File: tardis/montecarlo/packet_source.py [hunk 3]
```diff
@@ -110,9 +110,10 @@ class BlackBodySimpleSource(BasePacketSource):
     part.
     """
 
-    def create_packets(self, T, no_of_packets, rng):
+    def create_packets(self, T, no_of_packets, rng, radius):
```
Comment (wkerzendorf) [inline tardis/montecarlo/packet_source.py:113]: add `docstring`

**Review by wkerzendorf — APPROVED** [review]
Approved with minor changes 😉

**Pull request merged**